src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -659,20 +660,20 @@ private void ReportAssignmentOperatorError(AssignmentExpressionSyntax node, Diag
 
         private static void ReportBinaryOperatorError(ExpressionSyntax node, DiagnosticBag diagnostics, SyntaxToken operatorToken, BoundExpression left, BoundExpression right, LookupResultKind resultKind)
         {
-            bool leftDefault = left.IsLiteralDefault();
-            bool rightDefault = right.IsLiteralDefault();
-            if ((operatorToken.Kind() == SyntaxKind.EqualsEqualsToken || operatorToken.Kind() == SyntaxKind.ExclamationEqualsToken))
+            bool leftDefaultOrNew = left.IsLiteralDefault() || left.IsTypelessNew();
+            bool rightDefaultOrNew = right.IsLiteralDefault() || right.IsTypelessNew();
+            if (operatorToken.Kind() == SyntaxKind.EqualsEqualsToken || operatorToken.Kind() == SyntaxKind.ExclamationEqualsToken)
             {
-                if (leftDefault && rightDefault)
+                if (leftDefaultOrNew && rightDefaultOrNew)
                 {
-                    Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnDefault, node, operatorToken.Text);
+                    Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnTypelessExpression, node, operatorToken.Text, left.Display, right.Display);
                     return;
                 }
             }
-            else if (leftDefault || rightDefault)
+            else if (leftDefaultOrNew || rightDefaultOrNew)
             {
-                // other than == and !=, binary operators are disallowed on `default` literal
-                Error(diagnostics, ErrorCode.ERR_BadOpOnNullOrDefault, node, operatorToken.Text, "default");
+                // other than == and !=, binary operators are disallowed on `default` literal or new()
```

Comment:
>or new() [](start = 94, length = 8)

I may have lost track, but I thought `new()` would be allowed in binary operators. Let's check LDM notes and add a PROTOTYPE accordingly.

Reply:
The message is incorrect. we allow all binary operators for new(). we'll fix,